zappa/core.py

```diff
@@ -2414,7 +2414,7 @@ def create_stack_template(
 
         # build a fresh template
         self.cf_template = troposphere.Template()
-        self.cf_template.add_description("Automatically generated with Zappa")
+        self.cf_template.set_description("Automatically generated with Zappa")
```

Comment:
Do we also need to update the requirements to force usage of the new troposphere?
If someone is using `troposphere==2.7.0`, will this still work?

Reply:
Yes, I agree that this would break if this PR ran on a deploy that used `troposphere==2.7.0`. The core issue imo is that `troposphere` isn't pinned to a version yet Zappa is relying on a version specific api from troposphere.

@povesma could add a change to the `requirements.txt` that pins the version of `troposphere` to a 3x major version?

Reply:
if we want to keep backwards compatibility why not do something like:
```
if hasattr(self.cf_template, 'add_description'):
    add_description
elif hasattr(self.cf_template, 'set_description'):
    set_description
```

Reply:
The above backwards compatibility change is not needed for troposphere v2.7.x as the change (from v2.4.2 2019-02-02) included both methods (with a deprecation warning on the set_* variant). Of course, the reason for v3.0.0 was to deprecate Python 2.x support and only support Python 3+ in case that factors into zappa's Python version support.